Static attention implementation

sxu opened the pull request:
Summary: Merge existing implementation for static shaped NPU where valid data are kept at the end of cache and MHA -> SHA rewrite is done.

Reviewed By: iseeyuan

Differential Revision: D69080741

iseeyuan reviewed (approved):
Related topic in https://github.com/pytorch/executorch/discussions/8228

sxu committed:
Static attention implementation (#8310)

Summary:

Merge existing implementation for static shaped NPU where valid data are kept at the end of cache and MHA -> SHA rewrite is done.

Reviewed By: iseeyuan

Differential Revision: D69080741
